doc/src/modules/solvers/solveset.rst

```diff
@@ -1,12 +1,463 @@
+
+ Solveset is designed to be independent of the assumptions on the
+ variable being solved for and instead, uses the ``domain`` argument to
+ decide the solver to dispatch the equation to, namely ``solveset\_real``
+ or ``solveset\_complex``. It's unlike the old ``solve`` which considers the
+ assumption on the variable.
+
+    >>> from sympy.solvers.solveset import solveset
+    >>> from sympy.abc import x
+    >>> from sympy import S
+    >>> solveset(x**2 + 1, x) # domain=S.Complexes is default
+    {-I, I}
+    >>> solveset(x**2 + 1, x, domain=S.Reals)
+    EmptySet()
+
+
+What are the general methods employed by solveset to solve an equation?
+-----------------------------------------------------------------------
+
+ Solveset uses various methods to solve an equation, here is a brief overview
+ of the methodology:
+
+ * The ``domain`` argument is first considered to know the domain in which
+   the user is interested to get the solution.
+
+ * If the given function is a relational (`>=`, `<=`, `>`, `<`), and the
+   domain is real, then `solve\_univariate\_inequality` and solutions are
+   returned. Solving for complex inequalities are not supported yet.
+
+
+ * Based on the ``domain``, the equation is dispatched to one of the two
+   functions ``solveset\_real`` or ``solveset\_complex``, which solves the
+   given equation in complex and real domain respectively.
+
+ * If the given function (equation) is a product of two or more functions,
+   like say `f = g*h`, then the solution to the given equation is the Union
+   of the solution of the equations `g = 0` and `h = 0`, if and only if both
+   `g` and `h` are finite for a finite input. So, the solution is build up
+   recursively.
+
+ * The function class is now checked if it's Trigonometric or Hyperbolic, then
+   the function ``\_solve\_real\_trig`` is called, which solves it by converting
+   it in terms of ``exp`` complex exponential form.
+
+ * The function is now checked if there is any instance of Piecewise
+   expression, if it is, then it's converted to explict expression and
+   set pairs and then solved recursively.
+
+ * The respective solvers now tries to invert the equation using the routines
+   ``invert\_real`` and ``invert\_complex``.
+
+ * After the invert, the equations are checked for radical or Abs (Modulus),
+   then the method ``\_solve\_radical`` tries to simplify the radical, by
+   removing it using techniques like squarring, cubing etc, and ``\_solve\_abs``
+   solves nested Modulus by considering the positive and negative variants,
+   iteratively.
+
+ * If none of the above method is successful, then method of polynomial is
+   used as follows:
+
+   - ``\_solve\_as\_rational`` is called, it's third argument is the
+     ``solveset\_solver`` which can either be ``solveset_real`` or
+     ``solveset\_complex`` based on these respective poly solvers
+     ``\_solve\_as\_poly\_real`` and ``\_solve\_as\_poly\_complex`` is called.
```

Comment:
This section doesn't make much sense to me. What are the first and second arguments? It's not clear how `_solve_as_rational`, `solveset_solver` (is that even a real function?), `solveset_real`, and `_solve_as_poly_real` fit together here.